### PR TITLE
moving the remaining engines unit tests over to pytest

### PR DESCRIPTION
**Pull request by garethgreenaway**
### What does this PR do?
moving the remaining engines unit tests over to pytest

### Merge requirements satisfied?
**[NOTICE] Bug fixes or features added to Salt require tests.**

- [ ] Docs
- [ ] Changelog - https://docs.saltproject.io/en/master/topics/development/changelog.html
- [ ] Tests written/updated

### Commits signed with GPG?
Yes

Please review [Salt's Contributing Guide](https://docs.saltproject.io/en/master/topics/development/contributing.html) for best practices.

See GitHub's [page on GPG signing](https://help.github.com/articles/signing-commits-using-gpg/) for more information about signing commits with GPG.

### CODE REVIEW

**Review by twangboy — DISMISSED** [review]

File: tests/pytests/unit/engines/test_libvirt_events.py [hunk 1]
```diff
@@ -0,0 +1,202 @@
...
+        mock_libvirt.VIR_DOMAIN_EVENT_ID_REBOOT = 1
+        mock_libvirt.VIR_STORAGE_POOL_EVENT_ID_LIFECYCLE = 0
+        mock_libvirt.VIR_STORAGE_POOL_EVENT_ID_REFRESH = 1
+        mock_libvirt.VIR_NODE_DEVICE_EVENT_ID_LIFECYCLE = 0
+        mock_libvirt.VIR_NODE_DEVICE_EVENT_ID_UPDATE = 1
+        yield mock_libvirt
+    finally:
+        patcher.stop()
+
+
```
Comment (s0undt3ch) [inline tests/pytests/unit/engines/test_libvirt_events.py:31]: Can we not use the `patch` as a decorator and instead use a with statement?

File: tests/pytests/unit/engines/test_libvirt_events.py [hunk 2]
```diff
@@ -0,0 +1,202 @@
...
+    """
+    Make sure the libvirt enum value to string works reliably with
+    elements with a sub prefix, eg VIR_PREFIX_SUB_* in this case.
+    """
+    # Test case with a sub prefix
+
+    assert libvirt_events._get_libvirt_enum_string("VIR_PREFIX_", 2) == "two"
+
+
+@patch("salt.engines.libvirt_events.libvirt", VIR_PREFIX_FOO=0, VIR_PREFIX_BAR_FOO=1)
```
Comment (s0undt3ch) [inline tests/pytests/unit/engines/test_libvirt_events.py:51]: Same as above.
Of you end up needing to patch the same thing over and over, use a fixture. You can then decorate the test function with `@pytest.mark.usefixture("the fixture name")`

File: tests/pytests/unit/engines/test_sqs_events.py [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+
+@pytest.fixture
+def mock_sqs():
+    try:
+        patcher = patch("salt.engines.sqs_events.boto.sqs")
+        mock_sqs = patcher.start()
+        yield mock_sqs
+    finally:
+        patcher.stop()
+
```
Comment (s0undt3ch) [inline tests/pytests/unit/engines/test_sqs_events.py:28]: Instead of patcher.start/stop you can do:
```python
with patch(.....) as foo:
    yield foo
```

**Review by s0undt3ch — CHANGES_REQUESTED** [review]
Mostly let's remove the patch decorators.

**Review by Ch3LL — CHANGES_REQUESTED** [review]
The slack tests are being skipped: https://jenkins.saltproject.io/job/pr-centos-7-x86_64-py3-pytest/job/PR-60761/1/testReport/tests.pytests.unit.engines/test_slack/

**Comment by garethgreenaway** [comment]
@Ch3LL The Slack engine is broken with later version of the Slackclient python library, probably a good idea to leave them disabled until the engine is fixed.

**Comment by Ch3LL** [comment]
I'm okay with that. Can you open an issue to track getting these tests re-enabled so we do not forget?

**Review by Ch3LL — APPROVED** [review]

**Pull request merged**